libffi-sys-rs/src/lib.rs

```diff
@@ -162,17 +177,31 @@ pub struct ffi_closure {
     pub user_data: *mut c_void,
 }
 
+/// Implements Debug manually since sometimes FFI_TRAMPOLINE_SIZE is too large
+impl Debug for ffi_closure {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ffi_closure")
+            .field("tramp", &&self.tramp[..])
+            .field("cif", &self.cif)
+            .field("fun", &self.fun)
+            .field("user_data", &self.user_data)
+            .finish()
+    }
+}
+
```

Comment:
Is this really necessary in this PR? If the intentions are to improve portability, I'd rather have the PR focus on _just_ that; anything not directly related to that should go in a separate PR, making it easier to review changes :smile:

Reply:
It is related to portability - it lets the code build on `powerpc-unknown-linux-gnu`, though that target has other problems.

Reply:
In that case, how does implementing Debug/etc make this code more portable? Is it because we're already deriving `Debug` in places and that doesn't work on powerpc?

Reply:
You can't `derive(Debug)` on structs with arrays that have more than 32 elements. On PowerPC `FFI_TRAMPOLINE_SIZE` is 40.

Reply:
Aaah gotcha, then we can keep the Debug implementation code. Thanks for clarifying! :smile: